internal/runner/rspec.go

```diff
@@ -63,64 +67,72 @@ func (r Rspec) GetFiles() ([]string, error) {
 	return files, nil
 }
 
-func (r Rspec) RetryCommand() (*exec.Cmd, error) {
-	words := []string{}
-
-	if r.RetryTestCommand == "" {
-		// use test command to build retry command if retry command is not provided
-		// remove all occurrences of "{{testExamples}}" from the test command and append "--only-failures"
-		splits, err := shellquote.Split(r.TestCommand)
-		if err != nil {
-			return nil, err
-		}
-		splits = slices.DeleteFunc(splits, func(n string) bool {
-			return n == "{{testExamples}}"
-		})
-		splits = slices.Insert(splits, len(splits), "--only-failures")
-		words = append(words, splits...)
+// Run executes the test command with the given test cases.
+// If retry is true, it will run the command using the retry test command,
+// otherwise it will use the test command.
+//
+// Error is returned if the command fails to run, exits prematurely, or if the
+// output cannot be parsed.
+//
+// Test failure is not considered an error, and is instead returned as a RunResult.
+func (r Rspec) Run(testCases []string, retry bool) (RunResult, error) {
+	var command string
+
+	if retry {
+		command = r.RetryTestCommand
 	} else {
-		splits, err := shellquote.Split(r.RetryTestCommand)
-		if err != nil {
-			return nil, err
-		}
-		words = append(words, splits...)
+		command = r.TestCommand
 	}
 
-	fmt.Println(shellquote.Join(words...))
-
-	cmd := exec.Command(words[0], words[1:]...)
-	cmd.Stderr = os.Stderr
-	cmd.Stdout = os.Stdout
-	return cmd, nil
-}
+	commandName, commandArgs, err := commandNameAndArgs(command, testCases)
+	if err != nil {
+		return RunResult{Status: RunStatusError}, fmt.Errorf("failed to build command: %w", err)
+	}
 
-// Command returns an exec.Cmd that will run the rspec command
-func (r Rspec) Command(testCases []string) (*exec.Cmd, error) {
-	commandName, commandArgs, err := r.commandNameAndArgs(testCases)
+	// Create a temporary file to store the JSON output of the rspec run.
+	// This is a temporary solution, until we have an option to specify the output file.
+	f, err := os.CreateTemp("", "rspec-*.json")
 	if err != nil {
-		return nil, err
+		return RunResult{Status: RunStatusError}, fmt.Errorf("failed to create temporary file for rspec output: %v", err)
 	}
-	fmt.Println(shellquote.Join(append([]string{commandName}, commandArgs...)...))
+	defer f.Close()
+	defer os.Remove(f.Name())
```

Comment:
Because `defer` statements are executed in reverse order, when the function returns this will remove the file and then close it. 

I don't think that causes problems on most Unix-like OSes, but something in the back of my head is telling me Windows might have problems with this order of operations...

You could either swap the order, or wrap them in the desired order within a single func:
```suggestion
	defer func() {
		f.Close()
		os.Remove(f.Name())
	}()
```